app/src/main/java/com/example/scoutconcordia/Activities/MapsActivity.java

```diff
@@ -353,56 +387,152 @@ public void initializeSearchBar(){
             }
         });
     }
+
     // onClick listener for when "Find your way" button is clicked
-    public void onFindYourWayButtonClick(final String destination){
+    private AlertDialog setOriginDialog(){
+        useMyLocation = true; // reset useMyLocation to false
         AlertDialog.Builder builder = new AlertDialog.Builder(this  );
         LayoutInflater inflater = this.getLayoutInflater();
-        View dialogView = inflater.inflate(R.layout.outdoor_buildings_diections_ui, null);
+        final View dialogView = inflater.inflate(R.layout.outdoor_buildings_diections_ui, null);
         builder.setView(dialogView);
 
         final AutoCompleteTextView startingLocation = dialogView.findViewById(R.id.starting_location);
-        RadioButton useCurrentLocationButton = dialogView.findViewById(R.id.useMyLocationButton);
-
+        travelOptionsMenu = dialogView.findViewById(R.id.travel_modes_nav_bar);
         ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_dropdown_item, locations);
         adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
-
         startingLocation.setAdapter(adapter);
 
+        ///when user clicks to enter starting point, uncheck useMyLocationButton if enabled
+        startingLocation.setOnTouchListener(new View.OnTouchListener() {
+            @Override
+            public boolean onTouch(View v, MotionEvent event) {
+                if(useMyLocation){ useMyLocationButtonClicked(dialogView); }
+                return false;
+            }
+        });
+
         startingLocation.setOnItemClickListener(new AdapterView.OnItemClickListener() {
             @Override
             public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
-
+                 startingPoint = startingLocation.getText().toString();
+                 if(!enableShuttle(dialogView, locationsInDifferentCampuses(startingPoint, destination))){
+                     setModeToWalkIfShuttleSelected();
+                 }
             }
         });
-        builder.create().show();
+        return builder.create();
+    }
+
+    private boolean locationsInDifferentCampuses(String startingPoint, String destination){
+        LatLng origin = locationMap.get(startingPoint);
+        LatLng dest = locationMap.get(destination);
+
+        float[] results = new float[1];
+        Location.distanceBetween(origin.latitude, origin.longitude, dest.latitude, dest.longitude, results);
+        return results[0] > 2000.0;
+    }
+
+    private boolean enableShuttle(View v, boolean enabled){
+        BottomNavigationItemView shuttleModeItem = v.findViewById(R.id.shuttle);
+        shuttleModeItem.setEnabled(enabled);
+        shuttleAvailable = enabled;
+        return enabled;
     }
 
-    // get directions button clicked in dialog for getting directions (in onFindYourWayButtonClick)
```

Comment:
Why was this comment removed

Reply:
It's because the method name was renamed and the new method name is self explanatory.